server/app.py

```diff
@@ -933,23 +932,17 @@ def customer_subscription_created(event):
 
 
 @celery.task(name="app.payment_intent_succeeded")
-def payment_intent_succeeded(event):
-    app.logger.info(f"Payment intent event: {event}")
-    payment_intent = event['data']['object']
-    invoice_id = payment_intent['invoice']
-    if invoice_id:
-        try:
-            invoice = stripe.Invoice.retrieve(invoice_id, expand=['subscription'])
-        except stripe.error.StripeError as e:
-            app.logger.error(f"Issue finding invoice for {payment_intent['id']} with message: {e}")
-            return # process can not continue without invoice and will need to be retriggered from stripe
+def payment_intent_succeeded(payment_intent_id):
+    payment_intent = stripe.PaymentIntent.retrieve(payment_intent_id, expand=["invoice.subscription"])
```

Comment:
**suggestion (non-blocking):** Might be overkill, but would it make sense to wrap this in a try/except, similar to how you handled the invoice retrieval previously?

Reply:
I don't think it's necessary. The reason why I had that in there previously is because I did run into situations where a payment_intent event object would have an invoice_id but the invoice wouldn't be available from stripe for whatever reason. I think this was simply a timing thing, whereas here we're calling with a payment_intent that we know has already been processed since it was the thing that was passed to us. There's probably a better way of explaining that, so we can talk about it in our 1:1 if you want to.

Reply:
All good, WFM!